tests/PHPStan/Analyser/data/array-filter.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace ArrayFilter;
```

Comment:
Separate test-cases for
- a string which is not callable
- A string to a class-method reference
- With a 3rg [`mode` arg](https://www.php.net/manual/en/function.array-filter.php)

Reply:
adapted. what do you think? unfortunately narrowing-down the return type is currently only working for built-in functions. not entirely sure if this is a normal phpstan limitation or not. and/or if I could add support for it 🤔 you'll understand what I mean if you take a look at the last block of tests